src/custom/utils/format.ts

```diff
@@ -144,6 +145,12 @@ export function formatSmartLocaleAware(...params: Parameters<typeof formatSmart>
   return formatSmart(value, decimalsToShow, { ...options, isLocaleAware: true, thousandSeparator: true })
 }
 
+export function formatSmartAmount(
+  value: CurrencyAmount<Currency> | Percent | BigNumber | Fraction | null | undefined
+): string | undefined {
+  return formatSmart(value, AMOUNT_PRECISION)
+}
```

Comment:
I would think we already have something for this already in the project, don't we?

Reply:
I haven't found :(